### PR TITLE
Refactor statement

### PR DESCRIPTION
**Pull request by hzxuzhonghu**
Signed-off-by: xuzhonghu <xuzhonghu@huawei.com>

### CODE REVIEW

File: pkg/scheduler/framework/statement.go [hunk 1]
```diff
@@ -254,8 +271,9 @@ func (s *Statement) Allocate(task *api.TaskInfo, hostname string) error {
 	// Update status in session
 	klog.V(3).Info("Allocating operations ...")
 	s.operations = append(s.operations, operation{
-		name: "allocate",
-		args: []interface{}{task, hostname},
+		name:   Allocate,
+		task:   task,
+		reason: hostname,
```
Comment (k82cn) [inline pkg/scheduler/framework/statement.go:276]: allcate reason is hostname?

File: pkg/scheduler/framework/statement.go [hunk 2]
```diff
@@ -25,22 +25,37 @@ import (
...
+const (
+	Evict = iota
+	Pipeline
+	Allocate
+)
+
+type operation struct {
+	name   Operation
+	task   *api.TaskInfo
+	reason string
```
Comment (k82cn) [inline pkg/scheduler/framework/statement.go:39]: let's use `args []string` to make it more general :)

File: pkg/scheduler/framework/statement.go (hunk 1 above)
Comment (hzxuzhonghu) [reply pkg/scheduler/framework/statement.go]: Removed. Actually only for evict it is used, for other ops useless.

**Pull request merged**